src/da.rs

```diff
@@ -108,25 +120,34 @@ impl DataAvailabilityLayer for CelestiaConnection {
                     match EpochJson::try_from(blob) {
                         Ok(epoch_json) => epochs.push(epoch_json),
                         Err(_) => {
-                            debug!("Could not parse epoch json for blob at height {}", height)
+                            DataAvailabilityError::DataRetrievalError(
+                                height,
+                                "Could not parse epoch json for blob".to_string(),
+                            );
                         }
                     }
                 }
                 Ok(epochs)
             }
-            Err(err) => Err(format!(
-                "Could not get height {} from DA layer: {}",
-                height, err
+            Err(err) => Err(DataAvailabilityError::DataRetrievalError(
+                height,
+                format!("Could not get epoch from DA layer: {}", err),
             )),
         }
     }
 
-    async fn submit(&self, epoch: &EpochJson) -> Result<u64, String> {
+    async fn submit(&self, epoch: &EpochJson) -> Result<u64, DeimosError> {
         debug! {"Posting epoch {} to DA layer", epoch.height};
-        // todo: unwraps (#11)
-        let data = serde_json::to_string(&epoch).unwrap();
-        let blob = Blob::new(self.namespace_id.clone(), data.into_bytes()).unwrap();
-        debug!("blob: {}", serde_json::to_string(&blob).unwrap());
+        
+        let data = serde_json::to_string(&epoch).map_err(|e| {
```

Comment:
non-blocking question: This is the only method that returns DeimosError. Would it make sense to wrap the DeimosError in a DataAvailabilityError instead of the other way around (which you do below)? Or is it better to stay DeimosError

Reply:
I agree, we should change it to DataAvailabilityError. This makes the use of the trait consistent and predictable for anyone who needs to implement the trait in the future, ty